src/Orleans.Runtime/Silo/SiloControl.cs

```diff
@@ -299,7 +292,7 @@ public Task<List<GrainId>> GetActiveGrains(GrainType grainType)
 
         public Task MigrateRandomActivations(SiloAddress target, int count)
         {
-            var statistics = GetLocalDetailedGrainStatistics();
+            var statistics = GetDetailedGrainStatistics().Result; // Task is always completed
```

Comment:
Must use ValueTask instead then

Reply:
That method is part of `ISiloControl` and other parts of the codebase expect it to be a Task, thats mostly used by mgmt grain which either way await all tasks in a Task.WhenAll so returning a ValueTask servers no purpose as they need to be cast to a Task eitherway. The change is not justified IMO especially knowing that this task will be completed when called eitherway

Reply:
A better way would be to extract that into a sync method and just call that, which GetDetailed... can call it also and wrap it in a task. I had this before, but removed it as it seemed to miniscule.

Reply:
> A better way would be to extract that into a sync method and just call that, which GetDetailed... can call it also and wrap it in a task. I had this before, but removed it as it seemed to miniscule.

Reverted back to it